src/event.js

```diff
@@ -91,6 +91,36 @@ function on( elem, types, selector, data, fn, one ) {
 	} );
 }
 
+function addEventGetter( unused, name ) {
+	Object.defineProperty( jQuery.Event.prototype, name, {
+		enumerable: true,
+		configurable: true,
+
+		get: function() {
+			var value, hook;
+
+			if ( this.originalEvent ) {
+				if ( hook = jQuery.event.propHooks[ name ] ) {
+					value = hook( this.originalEvent );
+				} else {
+					value = this.originalEvent[ name ];
+				}
+			}
+
+			return this[ name ] = value;
+		},
+
+		set: function( value ) {
+			Object.defineProperty( this, name, {
```

Comment:
There are only a few situations I can think of where this is going to be in a hot path, like mousemove or pointermove events. Low-frequency events aren't going to suffer in any of these approaches I'd think. @jbedard do you have some benchmarks we could look at, even something rough? Perhaps @topaxi has something he used on #2384.